### PR TITLE
[WIP] Test for v1beta1 snapshot API

### PR DESCRIPTION
**Pull request by sdodsley**
Only for testing against internal CI system

### CODE REVIEW

File: pure-csi/templates/rbac.yaml [hunk 1]
```diff
@@ -51,6 +51,9 @@ rules:
   - apiGroups: ["snapshot.storage.k8s.io"]
     resources: ["volumesnapshotcontents"]
     verbs: ["create", "get", "list", "watch", "update", "delete"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
```
Comment (pure-jliao) [inline pure-csi/templates/rbac.yaml:54]: might need this change in operator install.sh as well.

Comment (sdodsley) [reply pure-csi/templates/rbac.yaml:54]: Good catch. Done

File: pure-csi/templates/provisioner.yaml [hunk 2]
```diff
@@ -86,12 +86,11 @@ spec:
 {{- if not (.Capabilities.KubeVersion.GitVersion | regexMatch "gke") }}
         - name: csi-snapshotter
           {{- with .Values.csi.snapshotter.image }}
-          image: {{ .name | default "quay.io/k8scsi/csi-snapshotter" }}:v1.2.2
+          image: {{ .name | default "quay.io/k8scsi/csi-snapshotter" }}:v2.1.1
```
Comment (pure-jliao) [inline pure-csi/templates/provisioner.yaml:89]: as described in the release note https://github.com/kubernetes-csi/external-snapshotter/releases
from `v2.0.0` requries k8s 1.17 and above, I think we should keep `v1.2.2` for older k8s versions.

Comment (sdodsley) [reply pure-csi/templates/provisioner.yaml:89]: This is just to test if things work in 1.17. Will add version checks if they do

**Comment by sdodsley** [comment]
Closing this in favour of a more specific OpenShift 4.4 PR

**Pull request closed without merging**